libs/ui-toolkit/src/components/callout/callout.tsx

```diff
@@ -1,23 +1,62 @@
+import type { ReactNode } from 'react';
 import classNames from 'classnames';
 import { getIntentShadow, Intent } from '../../utils/intent';
 import type { IconName } from '../icon';
 import { Icon } from '../icon';
 
-export interface CalloutProps {
+interface CalloutRootProps {
   children?: React.ReactNode;
   title?: React.ReactElement | string;
   intent?: Intent;
-  iconName?: IconName;
   headingLevel?: 1 | 2 | 3 | 4 | 5 | 6;
 }
 
-export function Callout({
-  children,
-  title,
-  intent = Intent.Help,
-  iconName,
-  headingLevel,
-}: CalloutProps) {
+interface CalloutWithoutIcon extends CalloutRootProps {
+  iconName?: never;
+  icon?: never;
+}
+
+interface CalloutPropsWithIconName extends CalloutRootProps {
+  children?: React.ReactNode;
+  title?: React.ReactElement | string;
+  intent?: Intent;
+  headingLevel?: 1 | 2 | 3 | 4 | 5 | 6;
+  iconName: IconName;
+  icon?: never;
+}
+
+interface CalloutPropsWithIcon extends CalloutRootProps {
+  children?: React.ReactNode;
+  title?: React.ReactElement | string;
+  intent?: Intent;
+  headingLevel?: 1 | 2 | 3 | 4 | 5 | 6;
+  iconName?: never;
+  icon: ReactNode;
+}
+
+type CalloutProps =
+  | CalloutWithoutIcon
+  | CalloutPropsWithIconName
+  | CalloutPropsWithIcon;
+
+const getIconElement = (props: CalloutProps) => {
+  if (props.iconName) {
+    return (
+      <Icon
+        name={props.iconName}
+        className="fill-current ml-8 mr-16 mt-8"
+        size={20}
```

Comment:
Chore (blocking): This icon will normally convey semantic value, which makes this a good opportunity to improve accessibility. Let's include an optional `iconDescription` prop (optional because it's possible that the icon could be decorational, and can be ignored by assistive tech if so). If `iconDescription`, use it with the `aria-label` attribute. If not, add `aria-hidden="true"`. 

There's a good article in icon accessibility at https://fontawesome.com/docs/web/dig-deeper/accessibility

Reply:
nice, added to the icon props